src/string_search.h

```diff
@@ -274,111 +229,133 @@ inline uint8_t GetHighestValueByte(uint16_t character) {
 inline uint8_t GetHighestValueByte(uint8_t character) { return character; }
 
 
-template <typename PatternChar, typename SubjectChar>
-inline size_t FindFirstCharacter(Vector<const PatternChar> pattern,
-                              Vector<const SubjectChar> subject, size_t index) {
-  const PatternChar pattern_first_char = pattern[0];
+// Searches for a byte value in a memory buffer, back to front
+// Uses memrchr(3) on systems which support it, for speed
+// Falls back to a vanilla for loop on systems which don't
+inline const void* MemrchrFill(const void* haystack, uint8_t needle,
+                               size_t haystack_len) {
+#ifdef _GNU_SOURCE
```

Comment:
`_GNU_SOURCE` is only auto-defined on Linux.  That's just as well for OS X because I don't believe it has a `memrchr()` function (at least, 10.7 doesn't and that's our baseline) but FreeBSD and OpenBSD do.